swarm/api/manifest.go

```diff
@@ -33,7 +33,8 @@ import (
 )
 
 const (
-	ManifestType = "application/bzz-manifest+json"
+	ManifestType        = "application/bzz-manifest+json"
+	ResourceContentType = "application/bzz-resource"
```

Comment:
I'm still not sure I understand why their is a different mime type, what is the format of a "bzz-resource"? Isn't it just raw data (i.e. whatever has been set as the resource)?

Reply:
Please see https://github.com/ethersphere/go-ethereum/pull/204#discussion_r163073068

Reply:
I'll be honest, I'm completely lost. I thought the idea of a resource update was to prevent updating ENS on chain, but instead being able to "find" the latest update using a clever hashing scheme. What part of this needs a new mime type?

A mime type represents some specific encoding / formatting of data, what is the encoding / format of a "resource update"?

Reply:
The idea, as far as I understand, is this:

The goal, as far as I understand, is "transparent" retrieval of dynamic content. That is to say that in the traditional manifest structure one can nest manifests which in turn will retrieve dynamic content.

When a mutable resource is added, a manifest is also added to the swarm chunk store. The key of the manifest is then added to to ENS record.

The "path" of the manifest contains the key of the root chunk for the mutable resource. But since the mutable resources live in a different chunk store, and thus have to be handled differently by the API, some piece of data in the manifest has to exist in order to trigger this special treatment. The obvious choice if the existing manifest struct is to be used, although maybe not thoroughly semantically correct, is the Content-type. 

One alternative would be to have a separate manifest type for resource updates, which some sort of magic first bytes that could minimize the burden of checking for it during normal retrievals.

Generally I'm not such a fan of hooking extra checking code for *every* request being made to swarm. Probably this is just a bit too hastily added. Maybe I just take the whole thing out and we make a proper plan for implementing it.

Reply:
So are you saying that fetching data under the hash of a manifest is no longer guaranteed to always return the same data? I understand that is the case for ENS names, but not for hashed manifests? This seems to break a lot of assumptions about immutable data.

Anyway, if this is the case, why are we also introducing `bzz-resource://` if it works with `bzz://`?

Lastly, are we sure the code here _actually_ does what you say? I'm looking at it, but I see no code where existing manifests get new entries for resource updates.

Reply:
No, I can't say I'm sure. I realize this code was too rushed. I will remove all mutable resources code from the "normal" schemes and plan it more properly before having another go at it.

Reply:
@lmars not sure what you find mysteriolaus about this feature. This is basically the usecase for this entire thingy. Imagine i got `tron.eth` i put a manifest with the resource description at `tron.eth/news`. If i indicate the content-type a mutable resource update description then the swarm http api can handle this mime type. How? well obviously by looking up and following them, and display the latest.

the `bzz-resource` scheme is sure needed to for arbitrary historical lookups and posting.

Reply:
@zelig I guess I've always just made the assumption that fetching content under a hash like `bzz://<hash>/news` would _always_ return the same content (without needing to know that it actually points at something which changes). It seems my assumption is incorrect then?

Reply:
Another assumption I made was that mutable resources was about a faster ENS, so `tron.eth` could be updated off-chain. To change `tron.eth/news` you'd just make a resource update to `tron.eth` which modifies the hash of `/news`, rather than making a resource update directly to `tron.eth/news`.

Reply:
I think this discussion should probably not be here. Maybe we can have it during the presentation today @lmars?